specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,318 @@
+  ///
+  /// If the request is initiated by a cross-origin iframe without a user gesture, 
+  /// the request will be blocked and the `LaunchingExternalProtocol` event will not 
+  /// be raised.
+  /// \snippet SettingsComponent.cpp LaunchingExternalProtocol
+  HRESULT add_LaunchingExternalProtocol(
+      [in] ICoreWebView2LaunchingExternalProtocolEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with
+  /// `add_LaunchingExternalProtocol`.
+  HRESULT remove_LaunchingExternalProtocol(
+      [in] EventRegistrationToken token);
+  }
+
+/// Receives the `LaunchingExternalProtocol` event.
+[uuid(e5fea648-79c9-47aa-8314-f471fe627649), object, pointer_default(unique)] 
+interface ICoreWebView2LaunchingExternalProtocolEventHandler: IUnknown {
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2* sender,
+      [in] ICoreWebView2LaunchingExternalProtocolEventArgs* args);
+}
+
+/// Event args for `LaunchingExternalProtocol` event.
+[uuid(fc43b557-9713-4a67-af8d-a76ef3a206e8), object, pointer_default(unique)] 
+interface ICoreWebView2LaunchingExternalProtocolEventArgs: IUnknown {
+  /// The URI with the external protocol to be launched.
+
+  [propget] HRESULT Uri([out, retval] LPWSTR* value);
+
+  /// The origin initiating the external protocol launch.
+  /// The origin will be empty if the WebView2 navigated to the external protocol.
+
+  [propget] HRESULT InitiatingOrigin([out, retval] LPWSTR* value);
+
+  /// `TRUE` when the external protocol request was initiated through a user gesture.
+  ///
+  /// \> [!NOTE]\n\> Being initiated through a user gesture does not mean that user intended
+  /// to access the associated resource.
+
+  [propget] HRESULT IsUserInitiated([out, retval] BOOL* value);
+
+  /// `TRUE` when the external protocol request was initiated via a non-main frame that 
+  /// has a different origin then the owning top-level page. When this property is `TRUE`
+  /// this may indicate a potential security issue and it is advised to block this request.
+
+  [propget] HRESULT IsCrossOriginIframe([out, retval] BOOL* value);
+
+  /// The host may set this flag to cancel the external protocol launch.  If set to
+  /// `TRUE`, the external protocol will not be launched, and the default
+  /// dialog is not displayed. This property can be used to replace the normal 
+  /// handling of launching external protocols. 
+
+  [propget] HRESULT Cancel([out, retval] BOOL* value);
+
+  /// Sets the `Cancel` property. The default value is `FALSE`.
+
+  [propput] HRESULT Cancel([in] BOOL value);
+
+  /// Returns an `ICoreWebView2Deferral` object.  Use this operation to
+  /// complete the event at a later time.
+
+  HRESULT GetDeferral([out, retval] ICoreWebView2Deferral** value);
+}
+
+``` 
+## .NET and WinRT
+
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2LaunchingExternalProtocolEventArgs;

```

Comment:
Redundant declaration

Reply:
We have in the past declared the runtimeclasses, @david-risney do we have a standard for declaring the classes or not? I have seen both in other docs.

Reply:
This is probably boilerplate. It's common to forward-declare everything, so that you can define things in a logical order (rather than having to do it in a topological sort-friendly order).

Reply:
Tooling generates this. No change required.